### PR TITLE
Hughsie/udev device

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-udev-device.c [hunk 1]
```diff
@@ -816,6 +816,13 @@ fu_udev_device_unbind_driver(FuDevice *device, GError **error)
 	g_autoptr(GOutputStream) stream = NULL;
 
 	/* is already unbound */
+	if (priv->udev_device == NULL) {
+		g_set_error_literal(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_FOUND,
+				    "not yet initialized");
```
Comment (superm1) [inline libfwupdplugin/fu-udev-device.c:823]: Don't use "yet" here to better future proof other lifecycle use.

File: libfwupdplugin/fu-udev-device.c [hunk 2]
```diff
@@ -2241,6 +2255,8 @@ fu_udev_device_get_parent_with_subsystem(FuUdevDevice *self, const gchar *subsys
 	FuUdevDevicePrivate *priv = GET_PRIVATE(self);
 	g_autoptr(GUdevDevice) device_tmp = NULL;
 
+	if (priv->udev_device == NULL)
+		return NULL;
```
Comment (superm1) [inline libfwupdplugin/fu-udev-device.c:2259]: In 2.0.0 I wonder if it's worth adding an error to this function and changing the symbol.

File: libfwupdplugin/fu-udev-device.c (hunk 1 above)
Comment (hughsie) [reply libfwupdplugin/fu-udev-device.c:823]: valid, fixed.

File: libfwupdplugin/fu-udev-device.c (hunk 2 above)
Comment (hughsie) [reply libfwupdplugin/fu-udev-device.c:2261]: Agree. Fixed.

**Review by superm1 — APPROVED** [review]

**Pull request merged**